src/cache.c

```diff
@@ -117,14 +147,26 @@ void cache_free(struct cache *cache)
  * Store an entry in the cache
  *
  * This will also remove the least-recently-used items as necessary.
- * 
+ *
  * NOTE: doesn't check for duplicate cache entries
  */
 void cache_put(struct cache *cache, char *path, char *content_type, void *content, int content_length)
 {
     ///////////////////
     // IMPLEMENT ME! //
     ///////////////////
+
+    struct cache_entry *ce = alloc_entry(path, content_type, content, content_length);
+    dllist_insert_head(cache, ce);
+    hashtable_put(cache -> index, path, ce);
```

Comment:
pretty sure it should be `hashtable_put(cache -> index, cache -> path, ce);`

Reply:
it's the path that is being passed in... ce -> path throws errors.